python/ray/data/_internal/execution/operators/map_operator.py

```diff
@@ -399,8 +393,9 @@ def _map_task(
         A generator of blocks, followed by the list of BlockMetadata for the blocks
         as the last generator return.
     """
+    DataContext._set_current(data_context)
```

Comment:
is this fixing a bug?

Reply:
Previously propagating the DataContext is done in the `generate_map_rows_fn` series functions. I'm moving it here for this refactor. In the future, we should make this part of the task submission API in the physical operator.